pkg/ccl/backupccl/backup_test.go

```diff
@@ -1411,177 +1411,85 @@ into_db='restoredb', %s)`, encryptionOption), backupLoc1)
 	}
 }
 
-type inProgressChecker func(context context.Context, ip inProgressState) error
-
-// inProgressState holds state about an in-progress backup or restore
-// for use in inProgressCheckers.
-type inProgressState struct {
-	*gosql.DB
-	backupTableID uint32
-	dir, name     string
-}
-
-func (ip inProgressState) latestJobID() (jobspb.JobID, error) {
-	var id jobspb.JobID
-	if err := ip.QueryRow(
-		`SELECT job_id FROM crdb_internal.jobs ORDER BY created DESC LIMIT 1`,
-	).Scan(&id); err != nil {
-		return 0, err
-	}
-	return id, nil
-}
+func TestBackupJobUpdatesProgress(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	defer jobs.TestingSetProgressThresholds()()
 
-// checkInProgressBackupRestore will run a backup and restore, pausing each
-// approximately halfway through to run either `checkBackup` or `checkRestore`.
-func checkInProgressBackupRestore(
-	t testing.TB, checkBackup inProgressChecker, checkRestore inProgressChecker,
-) {
-	var allowResponse chan struct{}
-	var exportSpanCompleteCh chan struct{}
+	blockCh := make(chan struct{})
 	params := base.TestClusterArgs{}
 	knobs := base.TestingKnobs{
 		DistSQL: &execinfra.TestingKnobs{
 			BackupRestoreTestingKnobs: &sql.BackupRestoreTestingKnobs{
-				RunAfterExportingSpanEntry: func(_ context.Context, res *kvpb.ExportResponse) {
-					<-allowResponse
-					// If ResumeSpan is set to nil, it means that we have completed
-					// exporting a span and the job will update its fraction progressed.
-					if res.ResumeSpan == nil {
-						<-exportSpanCompleteCh
-					}
-				},
-				RunAfterProcessingRestoreSpanEntry: func(_ context.Context, _ *execinfrapb.RestoreSpanEntry) {
-					<-allowResponse
+				RunAfterExportingSpanEntry: func(_ context.Context, _ *kvpb.ExportResponse) {
+					<-blockCh
 				},
 			},
 		},
 		JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
 	}
 	params.ServerArgs.Knobs = knobs
 
-	const numAccounts = 100
-
-	ctx := context.Background()
-	_, sqlDB, dir, cleanup := backupRestoreTestSetupWithParams(t, multiNode, numAccounts,
```

Comment:
I like the idea that we can reduce the size of the test.  It could be worth a comment though on vaguely what conditions ensure that we get a set of of todoSpans that allow this test to record incremental progress in the way we expect.  Or perhaps a testing hook that gets passed the backup plan spec so that we could make some basic assertions about the number of backup spans assigned to the processor so that if it changes this test will be easier to debug.